shell/common/animator.cc

```diff
@@ -27,22 +27,27 @@ Animator::Animator(Delegate& delegate,
       waiter_(std::move(waiter)),
       last_begin_frame_time_(),
       dart_frame_deadline_(0),
+#if FLUTTER_SHELL_ENABLE_METAL
+      layer_tree_pipeline_(fml::MakeRefCounted<LayerTreePipeline>(2)),
```

Comment:
This isn't an issue on Metal, I believe because we're controlling when we actually present the frame so that it always happens before the transaction is committed.

Can we do that on OpenGL?

Reply:
presentsWithTransaction

Reply:
Yes - but the keyboard lag doesn't happen on metal. Without the wait added in this PR for metal, similar issues are seen but much worse.